core/templates/dev/head/player/PlayerServices.js

```diff
@@ -357,10 +378,30 @@ oppia.factory('oppiaPlayerService', [
     getGadgetPanelsContents: function() {
       return angular.copy(_exploration.skin_customizations.panels_contents);
     },
+    getInteractionThumbnailSrc: function(stateName) {
+      // TODO(sll): unify this with the 'choose interaction' modal in
+      // state_editor_interaction.html.
+      var interactionId = _exploration.states[stateName].interaction.id;
+      if (!interactionId) {
+        return '';
+      } else {
+        return (
+          '/extensions/interactions/' + interactionId + '/static/' +
+          interactionId + '.png');
+      }
+    },
+    getInteractionInstructions: function(stateName) {
+      var interactionId = _exploration.states[stateName].interaction.id;
+      if (!interactionId) {
+        return '';
+      } else {
+        return INTERACTION_SPECS[interactionId].instructions;
+      }
+    },
     isInteractionInline: function(stateName) {
       var interactionId = _exploration.states[stateName].interaction.id;
       return (
-        interactionId &&
+        !interactionId ||
```

Comment:
The change is what caught my eye, but the logic is what made me comment. I think it might be worth a one line comment that explicitly calls out that a null interaction is defined here to be inline, to avoid someone misunderstanding the meaning of the delicate logic in the future. Since this is outside the conversation skin, it doesn't make sense to explain why it's defined as inline. Perhaps instead, you could say that a null interaction is treated as an inline interaction so that the error message associated with it is displayed in the most compact way possible in whichever skin is active. That seems to be the purpose of the new logic.